src/protocol/channels.ts

```diff
@@ -2383,6 +2385,29 @@ export type ResponseFinishedOptions = {};
 export type ResponseFinishedResult = {
   error?: string,
 };
+export type ResponseSecurityDetailsParams = {};
+export type ResponseSecurityDetailsOptions = {};
+export type ResponseSecurityDetailsResult = {
+  value?: SecurityDetails,
```

Comment:
I think our type system thinks about it as a required field named `value?` :)

Reply:
I think TS is resolving the protocol.yml as intended:

```
securityDetails:
  returns:
    value: SecurityDetails?
```

a field named `value` that's `undefined | SecurityDetails`.

I suppose we could skip the wrapper `value` (for SecurityDetails since all fields are optional, but not for ServerAddr), but I wanted to maintain the same API that we have internally and keep SecurityDetails and ServerAddr consistent with eachother.

https://github.com/microsoft/playwright/blob/10a82f862c0c9b37ad8a35df55a1da42e3ccd3d0/src/server/network.ts#L364